n2y/database.py

```diff
@@ -79,9 +79,17 @@ def _init_icon(self, icon_notion_data):
 
     @property
     def parent(self):
-        if self.notion_parent["type"] == "workspace":
+        """
+        The parent should be a workspace, database ("block_id"), or page ("page_id").
+        """
+        type = self.notion_parent["type"]
+        if type == "workspace":
             return None
-        else:
+
+        if type == "block_id":
+            return self.client.get_database(self.notion_parent["block_id"])
```

Comment:
@russellkan are you sure this works? I'm surprised that "block_id" would be a valid database_id here ...

I think this is incorrect. See https://developers.notion.com/reference/parent-object

Reply:
You're right; it's not a valid database_id but it did pass without error. I took a look at the API reference and updated it so it returns the appropriate parent. Turns out the error was due to a database with a block parent, not another database.